front/src/applications/operationalStudies/components/NGE/NGE.tsx

```diff
@@ -75,6 +67,21 @@ const NGE = ({ dto, onOperation }: NGEProps) => {
     }
   }, [dto, ngeRootElement]);
 
+  useEffect(() => {
+    if (ngeRootElement && onOperation) {
+      const fnOpListener = (event: Event) => {
+        const customEvent = event as CustomEvent;
+        const op = customEvent.detail as NGEEvent;
+        if (onOperation) onOperation(op, ngeRootElement.netzgrafikDto);
+      };
+      ngeRootElement.addEventListener('operation', fnOpListener);
+      return () => {
+        ngeRootElement.removeEventListener('operation', fnOpListener);
+      };
+    }
+    return () => {};
+  }, [onOperation, ngeRootElement]);
```

Comment:
Why this change? What issue does this fix?

Reply:
Previously, the useEffect has no dependecy on the `onOperation` function (it was just at mount/unmount). 
So if its definition changed, it was not updated.  

It worked before because the `onOperation` was static, but it's not the case anymore. 
That's why I created here a dedicated `useEffect` that manages the operations listener, with the good dependency list.

https://react.dev/reference/react/useEffect#specifying-reactive-dependencies
>  Every reactive value used by your Effect’s code must be declared as a dependency

Reply:
Good point! Can we extract this in a separate commit and record the reason for the change in the commit message? Future me will be very thankful.

Reply:
Bumping this - please split in a separate commit with an explanation in the commit message.